packages/main/src/TimePicker.ts

```diff
@@ -540,6 +580,9 @@ class TimePicker extends UI5Element implements IFormInputElement {
 	}
 
 	_onkeydown(e: KeyboardEvent) {
+		if (isEnter(e) && this._internals?.form) {
```

Comment:
We could move this conditional on line 602 similar to how it is done for the `ui5-date-picker`.
That way the user won't be able to submit the form when the value help is open and interactions won't be mixed.